workflow/scripts/build_existing_chp_de.py

```diff
@@ -16,20 +16,30 @@
 import sys
 
 
-def clean_data(combustion, biomass, geodata):
+def clean_data(combustion, biomass, geodata, opts):
     """
     Clean the data and return a dataframe with the relevant information.
     PLZ is translated to longitude and latitude using the pyGeoDb data.
     """
     biomass.dropna(subset="Postleitzahl", inplace=True)
     biomass.rename(columns={'NameStromerzeugungseinheit': 'NameKraftwerk'}, inplace=True)
 
+    if 'I' not in opts:
+        # filter out industrial CHPs if industry is not included
+        industry = combustion.dropna(subset='Einsatzort')
+        to_drop = industry[industry.Einsatzort.str.contains('Industrie')].index
+        combustion.drop(index=to_drop, inplace=True)
+
     data = pd.concat([biomass, combustion], join='inner', ignore_index=True)
 
     # Get only CHP plants
     CHP_raw = data.query("ThermischeNutzleistung > 0").copy()
     CHP_raw.NameKraftwerk = CHP_raw.NameKraftwerk.fillna(CHP_raw.EinheitMastrNummer)
 
+    # delete duplicates - compromise only when KwkMastrNummer and Bruttoleistung is the same
+    CHP_raw = CHP_raw.sort_values(by=['KwkMastrNummer', 'Bruttoleistung'])
+    CHP_raw = CHP_raw.drop_duplicates(subset=['KwkMastrNummer', 'Bruttoleistung'], keep='last')
```

Comment:
As far as I can remember, different blocks could have different build years. Which assumption is applied here?

One could use the sorting to priorities the oldest or newest blocks. Currently it is not clear what the purpose of the sorting is.

Using average could be more accurate using a `.groupby('KwkMastrNummer')` but requires setting aggregation strategies for the relevant columns:

```py
strategies = {
    "ThermischeNutzleistung": "mean",
    "Energietraeger": "first",
    "Inbetriebnahmedatum": "mean",
    ...
}
CHP_raw.groupby("KwkMastrNummer").agg(strategies)
```